crates/common/tedge_config/src/sudo.rs

```diff
@@ -0,0 +1,53 @@
+use std::ffi::OsStr;
+use std::path::Path;
+use std::process::Command;
+use std::sync::Arc;
+
+use tracing::warn;
+
+use crate::TEdgeConfig;
+
+const SUDO: &str = "sudo";
+
+/// An object used to remember the user's sudo preference.
```

Comment:
The point is not only to remember user's preference (a bool would do), but also to enable root privileges.